hub/core/sample.py

```diff
@@ -72,7 +74,8 @@ def shape(self):
 
     @property
     def compression(self):
-        self._read_meta()
+        if self._compression is None:
```

Comment:
read meta gets called every time even if the sample is uncompressed?

Reply:
Compression type is actually part of meta. For uncompressed, `_read_meta()` would return early, so nothing happens.